<?xml version="1.0" encoding="UTF-8" ?>
<launch>
    <arg name="rviz" default="true"/>
    <arg name="device" default="cpu"/>
    <arg name="robot_init_xyz" default="[0., 0., 1.]"/>
    <arg name="robot_init_q" default="[0., 0., 0., 1.]"/>

    <node pkg="monoforce" type="fit_terrain_node" name="fit_terrain" output="screen">
        <rosparam subst_value="true">
            data_path: '/home/ruslan/data/robingas/data/22-08-12-cimicky_haj/marv/ugv_2022-08-12-15-18-34_trav/'
<!--            data_path: '/home/ruslan/data/traversability/marv/ugv_2022-08-12-16-37-03_trav/'-->
            cloud_ind: 0
            grid_res: 0.4
            d_min: 0.1
            d_max: 8.0
            damping: 10.
            elasticity: 10.
            friction: 0.9
            vel_tracks: [2., 2.]
            robot_mass: 10.
            robot_init_xyz: $(arg robot_init_xyz)
            robot_init_q: $(arg robot_init_q)
            use_terrain_cnn: false
            lr: 0.01
            total_sim_time: 10.0
            n_samples: 1000
            sample_len: 20
            n_train_iters: 1000
            device: $(arg device)
        </rosparam>
    </node>

    <node if="$(arg rviz)"
          pkg="rviz" type="rviz" name="rviz" args="-d $(dirname)/../config/rviz/train.rviz"/>
</launch>
